vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -29,10 +45,14 @@ def __init__(self,
         """
         super().__init__(disable_bonus_tokens=disable_bonus_tokens,
                          strict_mode=strict_mode)
+        self.use_flashinfer = use_flashinfer
+        if self.use_flashinfer:
+            assert not disable_bonus_tokens, \
```

Comment:
Could be just warning?

Reply:
Ideally, when disable_bonus_tokens, the bonus token should be -1.
However, if we use flashinfer and set disable_bonus_tokens, the bonus token will still have values (!= -1), which makes the results incorrect. I guess it might be better to just fail here?

Reply:
we can remove the `disable_bonus_token` path completely now that https://github.com/vllm-project/vllm/issues/4212 is fixed.

but if it's too much work let's just leave it as assert, that way "no failure" means user gets the experience we planned for them instead of missing a warning and getting subpar perf